hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/VolumeInfo.java

```diff
@@ -83,6 +88,24 @@ public VolumeInfo build() throws IOException {
     }
   }
 
+  private long getReserved(ConfigurationSource conf) {
+    Collection<String> reserveList = conf.getTrimmedStringCollection(
+        HDDS_DATANODE_DIR_DU_RESERVED);
+    for (String reserve : reserveList) {
+      String[] words = reserve.split(":");
+      if (words.length < 2) {
+        throw new IllegalArgumentException(
+            "Reserved space should config in pair");
```

Comment:
Throwing `IllegalArgumentException` instead of the `ArrayIndexOutOfBoundsException` still shuts down datanode during startup.

I think if the config is invalid, it should be ignored with a warning log message.